tests/foreman/api/test_classparameters.py

```diff
@@ -15,78 +15,197 @@
 
 @Upstream: No
 """
-
-from robottelo.decorators import run_only_on, stubbed, tier1, tier2, tier3
+import json
+from random import choice
+
+from fauxfactory import gen_boolean, gen_integer, gen_string
+from nailgun import entities
+from requests import HTTPError
+
+from robottelo import ssh
+from robottelo.config import settings
+from robottelo.datafactory import filtered_datapoint
+from robottelo.decorators import (
+    run_only_on,
+    skip_if_bug_open,
+    stubbed,
+    tier1,
+    tier2,
+)
 from robottelo.test import APITestCase
 
 
+@filtered_datapoint
+def valid_sc_parameters_data():
+    """Returns a list of valid smart class parameter types and values"""
+    return [
+        {
+            u'sc_type': 'string',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'boolean',
+            u'value': choice(['0', '1']),
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_integer(min_value=1000),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': -123.0,
+        },
+        {
+            u'sc_type': 'array',
+            u'value': "['{0}', '{1}', '{2}']".format(
+                gen_string('alpha'), gen_integer(), gen_boolean()),
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': '{{"{0}": "{1}"}}'.format(
+                gen_string('alpha'), gen_string('alpha')),
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': 'name=>XYZ',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': '{"name": "XYZ"}',
+        },
+    ]
+
+
+@filtered_datapoint
+def invalid_sc_parameters_data():
+    """Returns a list of invalid smart class parameter types and values"""
+    return [
+        {
+            u'sc_type': 'boolean',
+            u'value': gen_string('alphanumeric'),
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': gen_string('alpha'),
+        },
+        {
+            u'sc_type': 'array',
+            u'value': '0',
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': 'a:test',
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': '{a:test}',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': gen_string('alpha'),
+        },
+    ]
+
+
 class SmartClassParametersTestCase(APITestCase):
     """Implements Smart Class Parameter tests in API"""
 
+    @classmethod
+    def setUpClass(cls):
+        super(SmartClassParametersTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        cls.host = entities.Host().search(
+            query={'search': 'name=' + cls.host_name}
+        )[0]
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = entities.Environment().search(
```

Comment:
I wonder if we should add assertions at this level just to make sure that things are in place before we run the actual tests? For example, if you cannot find the puppet environment `production`, the entire class would fail and we wouldn't really know at a glance what the problem was. This would apply to all other calls that attempt to fetch an entity in the `setUpClass` method. I don't honestly think we currently do this, but it could become something to be discussed as a 'best practice' for future development perhaps?